core/events/events_block_field_intermediate_change.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Class for an event representing an intermediate change to a block's field's
+ * value.
+ *
+ * @class
+ */
+import * as goog from '../../closure/goog/goog.js';
+goog.declareModuleId('Blockly.Events.BlockFieldIntermediateChange');
+
+import type {Block} from '../block.js';
+import * as registry from '../registry.js';
+import {Workspace} from '../workspace.js';
+
+import {BlockBase, BlockBaseJson} from './events_block_base.js';
+import * as eventUtils from './utils.js';
+
+/**
+ * Notifies listeners when the value of a block's field has changed but the
+ * change is not yet complete, and is expected to be followed by a block change
+ * event.
+ */
+export class BlockFieldIntermediateChange extends BlockBase {
+  override type = eventUtils.BLOCK_FIELD_INTERMEDIATE_CHANGE;
+
+  // Intermediate events do not undo or redo.
```

Comment:
Can you add some info here about why intermediate events don't record undo? I think we'll be able to dig up your docs about this in the future, but might be nice to just give future people a brief summary!

Reply:
Done.